runtime/rastrace/trclog.c

```diff
@@ -1055,29 +1054,36 @@ getTrcBuf(UtThreadData **thr, UtTraceBuffer * oldBuf, int bufferType)
 	trcBuf->queueData.subscriptions = 0;
 	trcBuf->thr = NULL;
 
+	/*
+	 * Get a pointer to where the thread name begins in a record. We don't
+	 * just use (trcBuf->record.threadName) because threadName is declared
+	 * as an array of length one and we're almost certainly writing more
+	 * than one character there. Some compilers track the length of the
+	 * target for strcpy, etc.: we want to avoid warnings due to the
+	 * apparent lack of space (we've made sure there's enough room).
+	 */
+	recordThreadName = offsetof(UtTraceRecord, threadName) + (char *)(void *)&trcBuf->record;
```

Comment:
Do you need to cast twice here?  I don't think `(void *)` is required.

Reply:
Perhaps not, but I think it's safer with the extra cast as it should make it clear to the compiler that there's no relationship between the pointer `&trcBuf->record` and the result of the casts.

Consider an analogy with Java: You'll get a compile error trying to cast a `String` to an `Integer`, but you can write `(Integer)(Object)"oops"` which compiles just fine (though you'll get a `ClassCastException` at runtime). The cast to `Object` is comparable to the cast to `void *` (and there's no exception in C).

Reply:
I expect this particular part of the change will be unnecessary once eclipse/omr#5444 is merged.